projects/smoke_bts7040/src/main.c

```diff
@@ -0,0 +1,111 @@
+// A simple smoke test for BTS7040
+
+// Periodically take reading from user selected channels and log the result
+// Configurable items: wait time, FRONT or REAR power distro selection, channels to be tested
+
+#include "bts7040_load_switch.h"
+#include "controller_board_pins.h"
+#include "current_measurement_config.h"
+#include "gpio.h"
+#include "i2c.h"
+#include "interrupt.h"
+#include "log.h"
+#include "mux.h"
+#include "pca9539r_gpio_expander.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+// experimentally determined value: scaling factor of 670/1160 ~ 0.578 ohms, bias of -8 mA
+#define SMOKE_BTS7040_SENSE_RESISTOR 1160
+#define SMOKE_BTS7040_BIAS (-8)
```

Comment:
These values were experimentally determined for the BTS7200s, but the BTS7040s are different and we haven't done any experimentation with them. In the future we'll use this smoke test to calibrate what these values should be. I'd say that to match the power distribution project's values (found in current_measurement.h), set the resistor to 1210 and the bias to 0, and change the comment to say that these parameters need to be calibrated.